Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM golang:1.7.4
+
+EXPOSE 8080
+RUN \
+  curl -fsSLO https://get.docker.com/builds/Linux/x86_64/docker-latest.tgz && \
+  tar --strip-components=1 -xvzf docker-latest.tgz -C /usr/local/bin
+RUN mkdir -p /go/src/go.uber.org/sally
+WORKDIR /go/src/go.uber.org/sally
+ADD glide.yaml glide.lock /go/src/go.uber.org/sally/
+RUN go get -v github.com/Masterminds/glide && glide install
+RUN go get -v github.com/golang/lint/golint github.com/kisielk/errcheck honnef.co/go/staticcheck/cmd/staticcheck
```

Comment:
Same thing, done for caching, using changes in glide.yaml or glide.lock for the cache invalidation. It might be better actually to vendor these tools directly.